crates/iceberg/src/expr/predicate.rs

```diff
@@ -55,6 +60,24 @@ impl<T, const N: usize> LogicalExpression<T, N> {
     }
 }
 
+impl<T: Bind, const N: usize> Bind for LogicalExpression<T, N>
+where
+    T::Bound: Sized,
+{
+    type Bound = LogicalExpression<T::Bound, N>;
+
+    fn bind(self, schema: SchemaRef, case_sensitive: bool) -> Result<Self::Bound> {
+        let mut outputs: [Option<Box<T::Bound>>; N] = array_init(|_| None);
+        for (i, input) in self.inputs.into_iter().enumerate() {
+            outputs[i] = Some(Box::new(input.bind(schema.clone(), case_sensitive)?));
+        }
+
+        // It's safe to use `unwrap` here since they are all `Some`.
```

Comment:
A bit confused. Why we need to create `[Some(T)]` and unwrap them one by one?

Reply:
Because without this, we can't initialize the array.

Reply:
How about using `MaybeUninit`?

Reply:
My first version uses `MaybeUninit`, but it involves `unsafe` codes. I think we should avoid `unsafe` as much as possible.

Reply:
Makes sense.